Uplift/Views/Cells/FacilitiesDropdownCell.swift

```diff
@@ -9,136 +9,96 @@ import UIKit
-        headerNameLabel.font = ._16MontserratRegular
-        headerNameLabel.textColor = .black
-        headerNameLabel.textAlignment = .left
-        headerView.addSubview(headerImageView)
-
-        let isFacilityOpen = getIsFacilityOpen()
-        headerOpenLabel.font = ._16MontserratRegular
-        headerOpenLabel.text = isFacilityOpen ? ClientStrings.CommonStrings.open : ClientStrings.CommonStrings.closed
-        headerOpenLabel.textColor = isFacilityOpen ? .accentOpen : .accentClosed
-        headerOpenLabel.textAlignment = .center
-        headerView.addSubview(headerOpenLabel)
     }
 
-    func getIsFacilityOpen() -> Bool {
-        let todaysDate = Date()
-        let dayOfWeek = todaysDate.getIntegerDayOfWeekToday()
-        for detail in facility.details {
-            if !detail.times.isEmpty {
-                for time in detail.times {
-                    if time.dayOfWeek == dayOfWeek {
-                        for timeRange in time.timeRanges {
-                            if todaysDate > timeRange.openTime && todaysDate < timeRange.closeTime {
-                                return true
-                            }
-                        }
-                    }
-                }
-            }
+    static func getHeight(for facilityDetail: FacilityDetail) -> CGFloat {
+        var height: CGFloat = 0
+        switch facilityDetail.detailType {
+        case .equipment:
+            height = EquipmentListCell.getHeight(models: facilityDetail.getEquipmentCategories())
+        case .hours:
+            height = FacilitiesHoursCell.getHeight(for: facilityDetail)
+        case .prices:
+            height = PriceInformationCell.getHeight(for: facilityDetail.items)
+        case .subfacilities:
+            height = MiscellaneousInfoCell.getHeight(for: facilityDetail.subfacilities)
         }
-        return false
+        return height
     }
 
-    func setupConstraints() {
-        let headerImageViewSize: CGFloat = 24
-        let headerImageViewLeadingOffset: CGFloat = 25
-        let headerImageViewNameLabelSpacing: CGFloat = 9
-        let openLabelTrailingOffset: CGFloat = -45
-
-        headerImageView.snp.makeConstraints { make in
-            make.height.width.equalTo(headerImageViewSize)
-            make.leading.equalToSuperview().offset(headerImageViewLeadingOffset)
-            make.centerY.equalToSuperview()
-        }
-
-        headerNameLabel.snp.makeConstraints { make in
-            make.leading.equalTo(headerImageView.snp.trailing).offset(headerImageViewNameLabelSpacing)
-            make.trailing.equalTo(headerOpenLabel.snp.leading).inset(headerImageViewNameLabelSpacing)
-            make.centerY.equalToSuperview()
-        }
-
-        headerOpenLabel.snp.makeConstraints { make in
-            make.centerY.equalToSuperview()
-            make.trailing.equalToSuperview().offset(openLabelTrailingOffset)
+    static func getHeights(for facility: Facility) -> CGFloat {
```

Comment:
can we rename to `getHeight`? since (i believe) in practice it's used to get the height of the one entire cell, despite the fact that there's cells nested inside, i.e. just returning one height.
or explicitly `getHeightForFacility` since you've already used `getHeight`?